WooCommerce/WooCommerceTests/ViewRelated/Dashboard/DashboardViewModelTests.swift

```diff
@@ -403,6 +405,185 @@ final class DashboardViewModelTests: XCTestCase {
+        userDefaults[.completedAllStoreOnboardingTasks] = true
+
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, featureFlags: featureFlagService, userDefaults: userDefaults)
+
+        mockLoadDashboardCards()
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertFalse(viewModel.dashboardCards.first(where: {$0.type == .onboarding })!.enabled)
+    }
+
+    func test_dashboard_cards_is_loaded_from_storage_if_they_exist() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isDynamicDashboardM2Enabled: false)
+        let storage = MockStorageManager()
+        let insertOrder = Order.fake().copy(siteID: sampleSiteID)
+        storage.insertSampleOrder(readOnlyOrder: insertOrder)
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, storageManager: storage, featureFlags: featureFlagService)
+
+        let storedCards = [DashboardCard(type: .onboarding, availability: .show, enabled: true),
+                           DashboardCard(type: .performance, availability: .show, enabled: true),
+                           DashboardCard(type: .topPerformers, availability: .show, enabled: true)]
+
+        mockLoadDashboardCards(withStoredCards: storedCards)
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertEqual(viewModel.dashboardCards, storedCards)
+    }
+
+    func test_dashboard_cards_respects_existing_ordering_from_saved_cards() {
+        // Given
+        let featureFlagService = MockFeatureFlagService(isDynamicDashboardM2Enabled: false)
+        let storage = MockStorageManager()
+
+        // Add order so that analytics cards are enabled
+        let insertOrder = Order.fake().copy(siteID: sampleSiteID)
+        storage.insertSampleOrder(readOnlyOrder: insertOrder)
+        let viewModel = DashboardViewModel(siteID: sampleSiteID, stores: stores, storageManager: storage, featureFlags: featureFlagService)
+
+        let storedCards = [DashboardCard(type: .topPerformers, availability: .show, enabled: true),
+                           DashboardCard(type: .onboarding, availability: .show, enabled: true),
+                           DashboardCard(type: .performance, availability: .show, enabled: true)]
+
+        mockLoadDashboardCards(withStoredCards: storedCards)
+
+        // When
+        viewModel.refreshDashboardCards()
+
+        // Then
+        waitUntil {
+            viewModel.dashboardCards.isNotEmpty
+        }
+
+        XCTAssertEqual(viewModel.dashboardCards[0], storedCards[0])
+        XCTAssertEqual(viewModel.dashboardCards[1], storedCards[1])
+        XCTAssertEqual(viewModel.dashboardCards[2], storedCards[2])
```

Comment:
Nit: Comparing the two arrays should be sufficient too.

Reply:
👍🏼 at first i wanted it to just be more explicit.

updated in https://github.com/woocommerce/woocommerce-ios/pull/12842/commits/e32160138974c206c3676fbcab9f70eb04ed421d